maebeale/mutual-aid
# Add basic support for docker

The `docker-compose` file is able to launch the rails application and
connect it to a postgres database, also running in docker. However, no other services are
working.

In particular, there is no ability to send emails at this time.
This means that I have not been able to create new user accounts, so much
of the core functionality of the application has not been tested.

This creates a new configuration which loads information from docker
environment variables.

To test this on your machine, you must
1. Have docker-compose installed and
2. You will need to generate a new `SECRET_KEY_BASE`, using the `rake secret` command.
3. The output of that command should be placed in a `.env` file in the
`docker/development/` folder. The `.env` should look like this:
```bash
SECRET_KEY_BASE=e0517d0887e68ebc518600...
```
4. That should be everything you need to get started. Just run
`docker-compose up --build` in the `docker/development/` directory, and
docker will build & launch the project.
5. You should be able to access the project at http://localhost:3000/
